From 08e5164ac1907455bb85ee2efdba699e8a928b15 Mon Sep 17 00:00:00 2001
From: "Wu, Fan" <fanxx.wu@intel.com>
Date: Thu, 16 Apr 2015 09:52:32 +0800
Subject: [PATCH] usb/gadget: reset cdev->next_string_id

Reset cdev->next_string_id to dev->reset_string_id before enable
gadget to avoid overflow of string_id reallocation.

Change-Id: I2dadb7cdd70eba004c5d5c07593571b09f7d01c0
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-26111
Signed-off-by: Wu, Fan <fanxx.wu@intel.com>
---
 drivers/usb/gadget/android.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/usb/gadget/android.c b/drivers/usb/gadget/android.c
index c4f42a1f7f8a..bd33ebad738e 100644
--- a/drivers/usb/gadget/android.c
+++ b/drivers/usb/gadget/android.c
@@ -425,10 +425,13 @@ static int functionfs_ready_callback(struct ffs_data *ffs)
 {
 	struct android_dev *dev = _android_dev;
 	struct functionfs_config *config = ffs_function.config;
+	struct usb_composite_dev *cdev = dev->cdev;
 	int ret = 0;
 
 	mutex_lock(&dev->mutex);
 
+	cdev->next_string_id = dev->reset_string_id;
+
 	config->instances++;
 	config->data = ffs;
 	config->opened = true;
-- 
1.9.1

